faunadb/strconv.go

```diff
@@ -0,0 +1,107 @@
+package faunadb
+
+import (
+	"reflect"
+	"strings"
+)
+
+func printFn(fn interface{}) string {
+
+	val := reflect.ValueOf(fn)
+
+	t := reflect.TypeOf(fn)
+	var name string
+	var sbOpt strings.Builder
+	var sbArgs strings.Builder
+	if t.Kind() != reflect.Struct {
+		return "fnApply should only be of type struct"
+	}
+	for idx := 0; idx < t.NumField(); idx++ {
+		field := t.Field(idx)
+		if field.Name == "fnApply" {
+			continue
+		}
+		if idx == 1 {
+			name = field.Name
+		}
+
+		fv := val.Field(idx)
+		var v Expr
+		if fv.IsNil() {
+			v = NullV{}
+		} else {
+			v = wrap(fv.Interface())
+		}
+
+		reprVals := map[string]string{}
+		delim := ""
+		if tag, b := field.Tag.Lookup("faunarepr"); b {
+			tagVals := strings.Split(tag, ",")
+			for _, key := range tagVals {
+				pair := strings.Split(key, "=")
+				switch len(pair) {
+				case 1:
+					reprVals["fn"] = pair[0]
+				case 2:
+					reprVals[pair[0]] = pair[1]
+				}
+			}
+
+			switch reprVals["fn"] {
+			case "scopedfn":
+				if (v != NullV{}) {
+					name = "Scoped" + name
+					if sbArgs.Len() > 0 {
+						delim = ", "
+					}
+					sbArgs.WriteString(delim + v.String())
+				}
+			case "optfn":
+				if (v != NullV{}) {
+					optFnName := field.Name
+					if len(reprVals["name"]) != 0 {
+						optFnName = reprVals["name"]
+					}
+					arg := optFnName + "("
```

Comment:
There are other places like this in this function where concatenation could be replaced by the string build for efficient, as intended.